rest/functionsapitest/graphql_queries_test.go

```diff
@@ -0,0 +1,260 @@
+//  Copyright 2022-Present Couchbase, Inc.
+//
+//  Use of this software is governed by the Business Source License included
+//  in the file licenses/BSL-Couchbase.txt.  As of the Change Date specified
+//  in that file, in accordance with the Business Source License, use of this
+//  software will be governed by the Apache License, Version 2.0, included in
+//  the file licenses/APL2.txt.
+
+package functionsapitest
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/graphql-go/graphql"
+
+	"github.com/couchbase/sync_gateway/base"
+	"github.com/couchbase/sync_gateway/db"
+	"github.com/couchbase/sync_gateway/db/functions"
+	"github.com/couchbase/sync_gateway/rest"
+	"github.com/stretchr/testify/assert"
+)
+
+// Test for GraphQL Query Admin-Only
+func TestGraphQLQueryAdminOnly(t *testing.T) {
+	rt := rest.NewRestTesterForUserQueries(t, rest.DbConfig{
+		GraphQL:       &kTestGraphQLConfig,
+		UserFunctions: &kTestGraphQLUserFunctionsConfig,
+	})
+	if rt == nil {
+		return
+	}
+	defer rt.Close()
+
+	t.Run("AsAdmin - getUser", func(t *testing.T) {
+		response := rt.SendAdminRequest("POST", "/db/_graphql", `{"query": "query($id:ID!){ getUser(id:$id) { id , name } }" , "variables": {"id": 1}}`)
+		assert.Equal(t, 200, response.Result().StatusCode)
+		assert.Equal(t, `{"data":{"getUser":{"id":"1","name":"user1"}}}`, string(response.BodyBytes()))
+
+		queryParam := `query($id:ID!){ getUser(id:$id) { id , name } }`
+		variableParam := `{"id": 1}`
+		getRequestUrl := fmt.Sprintf("/db/_graphql?query=%s&variables=%s", queryParam, variableParam)
+		response = rt.SendAdminRequest("GET", getRequestUrl, "")
+		assert.Equal(t, 200, response.Result().StatusCode)
+		assert.Equal(t, `{"data":{"getUser":{"id":"1","name":"user1"}}}`, string(response.BodyBytes()))
+	})
```

Comment:
There is a third way to query that also needs to be tested: the body of the POST is in GraphQL syntax and the Content-Type is `application/graphql`. (See rest/functions_api.go:168.)

Reply:
Added the test.